### PR TITLE
Remove more //base dependencies from //sky/shell

### PR DESCRIPTION
**Pull request by abarth**
This patch hides the base::SingleThreadedTaskRunner, which pulls a big
switchover to ftl::WeakPtr and std::function.

### CODE REVIEW

**Comment by abarth** [comment]
@chinmaygarde

File: sky/shell/shell.cc [hunk 1]
```diff
@@ -115,19 +110,18 @@ Shell::~Shell() {}
 void Shell::InitStandalone(std::string icu_data_path) {
   TRACE_EVENT0("flutter", "Shell::InitStandalone");
 
-  int file_descriptor =
-      icu_data_path.size() != 0
-          ? HANDLE_EINTR(::open(icu_data_path.data(), O_RDONLY))
-          : -1;
+  int icu_fd = icu_data_path.size() != 0
```
Comment (chinmaygarde) [inline sky/shell/shell.cc]: Maybe use the new `UniqueFD` wrapper here.

**Comment by chinmaygarde** [comment]
lgtm

File: sky/shell/shell.cc (hunk 1 above)
Comment (abarth) [inline sky/shell/shell.cc]: Done.

**Pull request merged**